code/game/objects/items/devices/PDA/apps/messenger.dm

```diff
@@ -89,6 +89,8 @@
                 U << browse(null, "window=pda")
                 return
         if("Message")
+            overlays.len = 0  // replying to a message from chat clears alert
+            update_icon()
```

Comment:
You're in the proc of the messenger app's datum. You need to do P.overlays.len = 0 and P.update_icon() instead (and obviously move those lines below the one that defines P.